ctpolicy/ctpolicy.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctpolicy contains structs describing CT policy requirements and
+// corresponding logic.
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// GroupBuilder owns information on log group membership logic.
+type GroupBuilder interface {
+	GetName() string
+	GetDescription() string
+	IsMember(log *loglist.Log) bool
+	IsBase() bool // whether group covers all logs
+}
```

Comment:
Having both `GroupBuilder` and `LogGroup` seems to involve a lot of copying things around -- is there any way to combine / simplify them?

Reply:
Simplified to some extent, PTAL.

Reply:
Thanks, this is much easier to follow.

I wonder if we could go one step further though.  It might be even simpler to have drop the `GroupBuilder` interface, and shift `BuildGroup` to have a signature like:
```golang
func buildGroup(name string, isBase bool, included func(log *loglist.Log)bool, ll *loglist.LogList) LogGroupInfo
```

Then the `chromepolicy.go` file loses a bunch of lines, as does this file -- `baseGroupBuilder` evaporates and just gets replaced with calls like 
```golang
base := buildGroup("All-Logs", true, func(log *loglist.Log)bool{return true}, approved)
```

Reply:
(This falls under "things I might have done differently" rather than something that definitely needs changing, btw)